### DISCUSSION TITLE
best pc-speaker demo

### DISCUSSION THREAD

**Discussion by stsp** [dosemu2/dosemu2#2288] [General]
[girl.exe.gz](https://github.com/user-attachments/files/17304896/girl.exe.gz)
Attached is the pc-speaker demo
girl.exe (Beatles song). It never worked
on either dosbox or dosemu[1|2], but
today I've found it working on dosbox-staging.
You can try it yourself, and you can hardly
tell its a pc-speaker, not SB16!
Really amazing progress for dosbox-staging.
I wonder what other emulators can
run this. So if you want to experiment -
here it is attached. :)

**Comment by jharrison03** [comment]
Here's another classic PC speaker demo:

[mushroom.zip](https://github.com/user-attachments/files/17511428/mushroom.zip)

**Reply by stsp** [reply]
Yep, sounds just as cool! :)
Eventually we'll get PC-Speaker too...

**Comment by jschwartzenberg** [comment]
How are you running this? I tried simply running them in the latest DOSBox-Staging but neither of them seems to work here.

**Reply by stsp** [reply]
Its a very recent achievement.
It works for me only with dosbox-staging
from git.

**Reply by jschwartzenberg** [reply]
Ah that explains! I was using v0.81.2.

**Reply by jschwartzenberg** [reply]
I built 43f29f09318ebe0f0b6db035cfb3a82d99298aaa but same behavior on my end. Nothing happens when I run either of those executables inside dosbox-staging. When I press a key, the program quits.

**Reply by stsp** [reply]
Here is my dosbox build, which
successfully runs these demos.
[dosbox.gz](https://github.com/user-attachments/files/17529613/dosbox.gz)
If I use 0.81.2 from rpm, I get the
same as you do - silence.

**Comment by stsp** [comment]
It is possible to also play the song natively:
```
sox -r 9321 -e signed -b 8 -c 1 -t raw MUSHROOM.OVL mush.wav
aplay mush.wav
```

**Reply by jharrison03** [reply]
I remember this silly commercial back in the '80s.  This is the source --

https://www.youtube.com/watch?v=ZMXuISih7Lc
